education/src/hometask8/FileNavigator.java

```diff
@@ -0,0 +1,71 @@
+package hometask8;
+
+import java.util.*;
+
+public class FileNavigator {
+
+    private final Map<String, List<FileData>> fileRepository;
+
+    public FileNavigator(Map<String, List<FileData>> fileRepository) {
+        this.fileRepository = fileRepository;
+    }
```

Comment:
it is a little bit useless to get Map from the constructor, init it inside or at a variable declaration